src/idl_parser.cpp

```diff
@@ -3513,8 +3517,11 @@ bool FieldDef::Deserialize(Parser &parser, const reflection::Field *field) {
   } else if (IsFloat(value.type.base_type)) {
     value.constant = FloatToString(field->default_real(), 16);
   }
-  deprecated = field->deprecated();
-  required = field->required();
+  // clang-format off
+  presence = field->required() ? FieldDef::kRequired
+           : field->optional() ? FieldDef::kOptional
+                               : FieldDef::kDefault;
```

Comment:
Duplicated code (see L865).
Better to add an internal non-member function like this:
```cpp
FieldDef::Presence MakeFieldPresence(bool required, bool optional);
```

Reply:
Done